### ISSUE TITLE
Feature request: Support circular references

### ISSUE THREAD

**Issue by sjoerd222888** [khalidsalomao/SimpleHelpers.Net#9]
Hi

Circular references are not supported by your library. I think it would be possible to support this by using `PreserveReferencesHandling = PreserveReferencesHandling.All` in the Newtonsoft serializer settings.

But of course a whole new level of complexity would be added.

Here a simple unit test (nunit-framework) with an example:

    private class CircularObject
    {
        private CircularObject child;

        public CircularObject Child
        {
            get { return child; }
            set { child = value; }
        }

        private CircularObject parent;

        public CircularObject Parent
        {
            get { return parent; }
            set { parent = value; }
        }
    }

    [Test]
    public void CircularReferencesTest()
    {
        CircularObject clientModel = new CircularObject();
        CircularObject serverModel = new CircularObject(); //equal object (with respect to properties)
        //start using ObjectDiffPatch
        var snapshot = ObjectDiffPatch.Snapshot(clientModel);
        //make a change (this is a cyclic reference)
        clientModel.Child = new CircularObject();
        clientModel.Child.Parent = clientModel;
        //create diff
        var diff = ObjectDiffPatch.GenerateDiff(snapshot, clientModel);
        //apply diff on server side
        serverModel = ObjectDiffPatch.PatchObject(serverModel, diff.NewValues);
        //ensure everything is alright
        Assert.AreEqual(serverModel, serverModel.Child.Parent);
   }

**Comment by khalidsalomao** [comment]
@sjoerd222888 

Thanks for the suggestion!

Unfortunately to change the default serializer settings would be a breaking change.

So instead of changing the default rule, I will expose the serialization settings so you can change it.

```csharp
ObjectDiffPatch.DefaultSerializerSettings.PreserveReferencesHandling = 
    Newtonsoft.Json.PreserveReferencesHandling.All;
```

Note that since it is a static property, it will affect all operations.

What do you think about that?

**Comment by khalidsalomao** [comment]
Feature published in Nuget release 1.4.0

**Comment by sjoerd222888** [comment]
Works as expected. You can close the issue.

Thx.
